packages/wallet-connect/src/types.ts

```diff
@@ -1,7 +1,12 @@
 import { SignClientTypes } from '@walletconnect/types';
 
 export type WcAccount = `${string}:${string}:${string}`;
+
 export type PolkadotNamespaceChainId = `polkadot:${string}`;
+
 export interface WalletConnectConfiguration extends SignClientTypes.Options {
+  projectId: string;
```

Comment:
This is set as optional, I've set it here as required, because it is required

Reply:
Any reason you want to do that, when the upstream lib itself accepts the optional value?!

If this is breaking the walletconnect.init I'd suggest opening an issue with them so they can fix it in the lib itself.
redefining the property technically voids the type safety here since they might change the property name in the future versions and then we would still be able to upgrade the lib to new version with no type error at build time and then the init can break at runtime!

Reply:
If it is really necessary to make it required in this lib, then sth like this might work to keep the type safety (there might be some more concise solutions as well)

```
type SignClientTypes = Omit<SignClientTypes.Options, 'projectId'> & Required<Pick<SignClientTypes.Options, 'projectId'>>;

export interface WalletConnectConfiguration extends SignClientTypes {
  chainIds?: PolkadotNamespaceChainId[];
  optionalChainIds?: PolkadotNamespaceChainId[];
  onSessionDelete?: () => void;
}
```

Reply:
Project id is required according to documentation and my personal experience with it
https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/overview#walletconnect-v10-vs-v20

> since they might change the property name
if they change any prop name that we use in the code, it will be a breaking change regardless of what is in typescript, and we would need to update accordingly.
`this.client = await SignClient.init(this.config);` this line will definitely give a type error at build if they change "projectId" to "appId" fro example.

I did create a PR though https://github.com/WalletConnect/walletconnect-monorepo/pull/3435
Do we really need to wait for this? Maybe just add a TODO or create an issue with reference to this PR and if it goes through, then do the change